File: src/main/java/net/obesephoenix/koalakraftcrystals/crystals/OnyxCrystal.java
package net.obesephoenix.koalakraftcrystals.crystals;

import org.bukkit.ChatColor;
import org.bukkit.block.Biome;
import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffectType;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class OnyxCrystal extends Crystal {

    public OnyxCrystal() {
        super("onyx",
                "onyx_crystal",
                "http://textures.minecraft.net/texture/f5db564300f9cd66785200799bef89894dc3c36b972722b782cda7d9c928b191",
                "end",
                Arrays.asList("Slow Falling", "Invisibility"),
                ChatColor.DARK_PURPLE,
                ChatColor.LIGHT_PURPLE);
    }

    @Override
    public void grantEffects(Player player) {

    }

    @Override
    protected List<String> addLorePrefix(List<String> lore) {
        lore.add("");
        lore.add(ChatColor.DARK_PURPLE + "A pure black gem. Staring into");
        lore.add(ChatColor.DARK_PURPLE + "its depths you can't help but feel");
        lore.add(ChatColor.DARK_PURPLE + "an overwhelming sense of dread...");
        return lore;
    }

}

File: src/main/kotlin/net/obesephoenix/koalakraftcrystals/crystals/OnyxCrystal.kt
package net.obesephoenix.koalakraftcrystals.crystals

import net.obesephoenix.koalakraftcrystals.crystals.Crystal
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.block.Biome
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import java.util.*

class OnyxCrystal : Crystal(
    "onyx",
    "onyx_crystal",
    "http://textures.minecraft.net/texture/f5db564300f9cd66785200799bef89894dc3c36b972722b782cda7d9c928b191",
    "end",
    listOf("Slow Falling", "Invisibility"),
    ChatColor.DARK_PURPLE,
    ChatColor.LIGHT_PURPLE
) {
    override fun grantEffects(player: Player?): Boolean {
        val biomes = listOf(
            Biome.END_BARRENS, Biome.THE_END, Biome.THE_VOID, Biome.END_HIGHLANDS,
            Biome.END_MIDLANDS, Biome.SMALL_END_ISLANDS
        )
        if (biomes.contains(player!!.location.block.biome)) {
            player.addPotionEffect(
                PotionEffect(
                    PotionEffectType.SLOW_FALLING, 340, 0,
                    false, false, true
                )
            )
            player.addPotionEffect(
                PotionEffect(
                    PotionEffectType.INVISIBILITY, 340, 0,
                    false, false, true
                )
            )
            return true
        }
        return false
    }

    override fun addLorePrefix(lore: MutableList<String?>): MutableList<String?> {
        lore.add("")
        lore.add(ChatColor.DARK_PURPLE.toString() + "A pure black gem. Staring into")
        lore.add(ChatColor.DARK_PURPLE.toString() + "its depths you can't help but feel")
        lore.add(ChatColor.DARK_PURPLE.toString() + "an overwhelming sense of dread...")
        return lore
    }
}